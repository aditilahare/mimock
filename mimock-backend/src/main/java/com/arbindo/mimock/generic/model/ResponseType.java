package com.arbindo.mimock.generic.model;

public enum ResponseType {
    TEXTUAL_RESPONSE,
    BINARY_RESPONSE
}
