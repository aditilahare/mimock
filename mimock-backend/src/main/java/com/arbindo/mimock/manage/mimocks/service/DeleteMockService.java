package com.arbindo.mimock.manage.mimocks.service;

public interface DeleteMockService {

    boolean hardDeleteMockById(String mockId);

    boolean softDeleteMockById(String mockId);

    boolean deleteAllMocks();

    void flushDeletedMocks();
}
