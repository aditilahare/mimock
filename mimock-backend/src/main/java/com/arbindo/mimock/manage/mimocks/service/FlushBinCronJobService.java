package com.arbindo.mimock.manage.mimocks.service;

public interface FlushBinCronJobService {
    void FlushBin();
}
