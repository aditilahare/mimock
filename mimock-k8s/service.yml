apiVersion: v1
kind: Service
metadata:
  name: mimock-svc
spec:
  type: NodePort
  selector:
    app: mimock
  ports:
    - protocol: TCP
      port: 8080
      targetPort: 8080
      nodePort: 30001
