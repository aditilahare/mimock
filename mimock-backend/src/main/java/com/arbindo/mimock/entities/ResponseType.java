package com.arbindo.mimock.entities;

import com.arbindo.mimock.generic.model.TypeOfResponse;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Getter;

import javax.persistence.*;
import java.time.ZonedDateTime;

@Getter
@Entity
@Table(name = "response_types")
public class ResponseType {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Long id;

    @Column(name = "name", nullable = false, length = 20)
    @Schema(example = "Response type name", description = "Field to indicate the type of a response field")
    @Enumerated(EnumType.STRING)
    private TypeOfResponse name;


    @Column(name = "created_at", nullable = false)
    @Schema(description = "Creation Timestamp")
    private ZonedDateTime createdAt;

    @Column(name = "updated_at")
    @Schema(description = "Update Timestamp")
    private ZonedDateTime updatedAt;

    @Column(name = "deleted_at")
    @Schema(description = "Delete Timestamp")
    private ZonedDateTime deletedAt;
}