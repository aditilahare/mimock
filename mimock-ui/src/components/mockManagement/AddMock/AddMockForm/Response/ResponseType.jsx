import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import TextField from '@mui/material/TextField';
import Autocomplete from '@mui/material/Autocomplete';
import { getResponseTypes } from 'services/staticRecords/getResponseContentTypes.service';
import { ResponseTypeWrapper, TypeLabel } from './Response.style';

function ResponseType({ type, responseContentType, setResponseContentType }) {
	const [responseTypes, setResponseTypes] = useState(null);
	const [filteredResponseTypes, setFilteredResponseTypes] = useState([]);
	const [defaultResponseType, setDefaultResponseType] =
		useState(responseContentType);

	useEffect(() => {
		getResponseTypes().then((types) => {
			setResponseTypes(types);
		});
	}, []);

	useEffect(() => {
		if (type && responseTypes) {
			const filteredTypes = responseTypes[type];
			filteredTypes?.includes(defaultResponseType)
				? setDefaultResponseType(defaultResponseType)
				: setDefaultResponseType(filteredTypes[0]);
			setFilteredResponseTypes(filteredTypes);
		}
	}, [type, responseTypes]);

	return (
		<If condition={type && filteredResponseTypes}>
			<ResponseTypeWrapper data-testid='response-type-wrapper'>
				<TypeLabel data-testid='response-type-label'>Response Type</TypeLabel>
				<Autocomplete
					disablePortal
					data-testid='response-type-autocomplete'
					id='response-types'
					options={filteredResponseTypes}
					defaultValue={defaultResponseType}
					value={defaultResponseType}
					sx={{ width: 300 }}
					renderInput={(params) => (
						<TextField {...params} label='Response type' />
					)}
					onChange={(e) => {
						setResponseContentType(e.target.value);
					}}
				/>
			</ResponseTypeWrapper>
		</If>
	);
}

ResponseType.propTypes = {
	type: PropTypes.string.isRequired,
	responseContentType: PropTypes.string.isRequired,
	setResponseContentType: PropTypes.func.isRequired,
};

export default ResponseType;
