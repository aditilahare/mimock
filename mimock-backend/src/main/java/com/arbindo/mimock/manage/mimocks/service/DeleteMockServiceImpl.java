package com.arbindo.mimock.manage.mimocks.service;

import com.arbindo.mimock.common.services.EntityStatusService;
import com.arbindo.mimock.entities.EntityStatus;
import com.arbindo.mimock.entities.Mock;
import com.arbindo.mimock.repository.MocksRepository;
import com.arbindo.mimock.utils.ValidationUtil;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.extern.log4j.Log4j2;
import org.apache.logging.log4j.Level;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import javax.transaction.Transactional;
import java.time.ZonedDateTime;
import java.util.List;

@Service
@Log4j2
@Builder
@AllArgsConstructor
public class DeleteMockServiceImpl implements DeleteMockService {

    @Autowired
    private MocksRepository mocksRepository;

    @Autowired
    private EntityStatusService entityStatusService;

    @Autowired
    private MockManagementService mockManagementService;

    @Override
    public boolean hardDeleteMockById(String mockId) {
        if (ValidationUtil.isNotNullOrEmpty(mockId)) {
            try {
                Mock mock = mockManagementService.getMockById(mockId);
                if (mock != null) {
                    mocksRepository.delete(mock);
                    return true;
                }
            } catch (Exception e) {
                log.log(Level.DEBUG, e.getMessage());
                return false;
            }
        }
        log.log(Level.DEBUG, "Invalid Mock Id!");
        return false;
    }

    @Transactional
    @Override
    public boolean softDeleteMockById(String mockId) {
        if (ValidationUtil.isNotNullOrEmpty(mockId)) {
            try {
                Mock mock = mockManagementService.getMockById(mockId);
                if (mock != null) {
                    // Perform only soft delete i.e. Mark EntityStatus as DELETED
                    EntityStatus entityStatus = entityStatusService.getDeletedMockEntityStatus();
                    mock.setEntityStatus(entityStatus);
                    mock.setDeletedAt(ZonedDateTime.now());
                    mocksRepository.save(mock);
                    return true;
                }
            } catch (Exception e) {
                log.log(Level.DEBUG, e.getMessage());
                return false;
            }
        }
        log.log(Level.DEBUG, "Invalid Mock Id!");
        return false;
    }

    @Override
    public boolean deleteAllMocks() {
        try {
            mocksRepository.deleteAll();
            return true;
        } catch (Exception e) {
            log.log(Level.DEBUG, e.getMessage());
        }
        log.log(Level.DEBUG, "Unable to delete all mocks!");
        return false;
    }

    @Override
    public void flushDeletedMocks() {
        try {
            EntityStatus entityStatus = entityStatusService.getDeletedMockEntityStatus();
            ZonedDateTime thirtyDaysAgo = ZonedDateTime.now().minusDays(30);
            List<Mock> deletedMocks = mocksRepository.findAllByEntityStatusAndDeletedAt(entityStatus, thirtyDaysAgo);
            mocksRepository.deleteAll(deletedMocks);
            log.log(Level.INFO, "Flushed " + deletedMocks.size() + " mock(s)!");
            return;
        } catch (Exception e) {
            log.log(Level.DEBUG, e.getMessage());
        }
        log.log(Level.DEBUG, "Unable to flush deleted mocks!");
    }
}
