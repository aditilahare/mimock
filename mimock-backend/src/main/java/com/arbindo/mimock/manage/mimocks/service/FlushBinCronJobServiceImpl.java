package com.arbindo.mimock.manage.mimocks.service;

import com.arbindo.mimock.entities.PlatformSettings;
import com.arbindo.mimock.manage.platformsettings.service.PlatformSettingsService;
import lombok.extern.log4j.Log4j2;
import org.apache.logging.log4j.Level;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.scheduling.annotation.Scheduled;
import org.springframework.stereotype.Service;

@Service
@Log4j2
public class FlushBinCronJobServiceImpl implements FlushBinCronJobService {

    @Autowired
    PlatformSettingsService platformSettingsService;

    @Autowired
    DeleteMockService deleteMockService;

    @Override
    @Scheduled(cron = "${flush-bin-cron-expression}")
    public void flushBin() {
        PlatformSettings platformSettings = platformSettingsService.getDefaultPlatformSettings();
        if(!platformSettings.getIsFlushBinCronEnabled()){
            log.log(Level.INFO, "Flush Bin Cron Is Disabled. Skipping CRON scheduler job.");
            return;
        }
        log.log(Level.INFO, "Starting CRON scheduler job.");
        deleteMockService.flushDeletedMocks();
        log.log(Level.INFO, "Completed CRON scheduler job.");
    }
}
