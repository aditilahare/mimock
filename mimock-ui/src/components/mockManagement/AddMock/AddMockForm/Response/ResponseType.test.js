import React from 'react';
import { act, render, fireEvent } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { getResponseTypes } from 'services/staticRecords/getResponseContentTypes.service';
import ResponseType from './ResponseType';

jest.mock('services/staticRecords/getResponseContentTypes.service');

describe('ResponseType', () => {
	it('should render response types', async () => {
		getResponseTypes.mockResolvedValue({
			TEXTUAL_RESPONSE: ['text/plain', 'text/html'],
			BINARY_RESPONSE: ['application/octet-stream', 'application/pdf'],
		});

		let tree;

		await act(async () => {
			tree = render(
				<ResponseType
					type='TEXTUAL_RESPONSE'
					responseContentType='text/plain'
					setResponseContentType={jest.fn()}
				/>
			);
		});

		const { getByTestId, container, getByRole } = tree;

		expect(getByTestId('response-type-wrapper')).toBeInTheDocument();
		expect(getByTestId('response-type-label')).toHaveTextContent(
			'Response Type'
		);
		expect(getByRole('textbox')).toHaveValue('text/plain');

		expect(container).toMatchSnapshot();
	});

	it('should change response type', async () => {
		getResponseTypes.mockResolvedValue({
			TEXTUAL_RESPONSE: ['text/plain', 'text/html'],
			BINARY_RESPONSE: ['application/octet-stream', 'application/pdf'],
		});

		let tree;

		await act(async () => {
			tree = render(
				<ResponseType
					type='BINARY_RESPONSE'
					responseContentType='text/plain'
					setResponseContentType={jest.fn()}
				/>
			);
		});

		const { getByTestId, container, getByRole } = tree;

		expect(getByTestId('response-type-wrapper')).toBeInTheDocument();
		expect(getByTestId('response-type-label')).toHaveTextContent(
			'Response Type'
		);
		expect(getByTestId('response-type-autocomplete')).toBeInTheDocument();
		const autoComplete = getByRole('textbox');
		expect(autoComplete).toHaveValue('application/octet-stream');

		await act(async () => {
			fireEvent.click(autoComplete);
			userEvent.type(autoComplete, 'application/pdf');
			fireEvent.change(autoComplete, {
				target: {
					value: 'application/pdf',
				},
			});
		});

		expect(autoComplete).toHaveValue('application/pdf');

		expect(container).toMatchSnapshot();
	});
});
