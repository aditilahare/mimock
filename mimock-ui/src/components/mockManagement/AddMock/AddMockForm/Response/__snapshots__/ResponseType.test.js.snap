// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResponseType should change response type 1`] = `
<div>
  <div
    class="flex items-center justify-start mt-2 mb-10 p-2 align-middle w-full"
    data-testid="response-type-wrapper"
  >
    <div
      class="font-sans text-gray-600 mx-4 text-justify font-semibold"
      data-testid="response-type-label"
    >
      Response Type
    </div>
    <div
      aria-expanded="true"
      aria-owns="response-types-listbox"
      class="MuiAutocomplete-root Mui-focused MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon css-viou3o-MuiAutocomplete-root"
      data-testid="response-type-autocomplete"
      role="combobox"
    >
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled Mui-focused css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="true"
          for="response-types"
          id="response-types-label"
        >
          Response type
        </label>
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth Mui-focused MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-autocomplete="list"
            aria-controls="response-types-listbox"
            aria-invalid="false"
            autocapitalize="none"
            autocomplete="off"
            class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
            id="response-types"
            spellcheck="false"
            type="text"
            value="application/pdf"
          />
          <div
            class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
          >
            <button
              aria-label="Clear"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-clearIndicator css-1glvl0p-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-clearIndicator"
              tabindex="-1"
              title="Clear"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                data-testid="CloseIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-label="Close"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator MuiAutocomplete-popupIndicatorOpen css-113ntv0-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
              tabindex="-1"
              title="Close"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-1z7n62"
            >
              <span>
                Response type
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      class="MuiAutocomplete-popper MuiAutocomplete-popperDisablePortal css-p8t6er-MuiAutocomplete-popper"
      data-popper-escaped=""
      data-popper-placement="bottom"
      data-popper-reference-hidden=""
      role="presentation"
      style="position: absolute; top: 0px; left: 0px; width: 0px; margin: 0px; transform: translate(0px, 0px);"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAutocomplete-paper css-9e5uuu-MuiPaper-root-MuiAutocomplete-paper"
      >
        <ul
          aria-labelledby="response-types-label"
          class="MuiAutocomplete-listbox css-b7k0tb-MuiAutocomplete-listbox"
          id="response-types-listbox"
          role="listbox"
        >
          <li
            aria-disabled="false"
            aria-selected="false"
            class="MuiAutocomplete-option"
            data-option-index="0"
            id="response-types-option-0"
            role="option"
            tabindex="-1"
          >
            application/pdf
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;

exports[`ResponseType should render response types 1`] = `
<div>
  <div
    class="flex items-center justify-start mt-2 mb-10 p-2 align-middle w-full"
    data-testid="response-type-wrapper"
  >
    <div
      class="font-sans text-gray-600 mx-4 text-justify font-semibold"
      data-testid="response-type-label"
    >
      Response Type
    </div>
    <div
      aria-expanded="false"
      class="MuiAutocomplete-root MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon css-viou3o-MuiAutocomplete-root"
      data-testid="response-type-autocomplete"
      role="combobox"
    >
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="true"
          for="response-types"
          id="response-types-label"
        >
          Response type
        </label>
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-autocomplete="list"
            aria-invalid="false"
            autocapitalize="none"
            autocomplete="off"
            class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
            id="response-types"
            spellcheck="false"
            type="text"
            value="text/plain"
          />
          <div
            class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
          >
            <button
              aria-label="Clear"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-clearIndicator css-1glvl0p-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-clearIndicator"
              tabindex="-1"
              title="Clear"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                data-testid="CloseIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-label="Open"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
              tabindex="-1"
              title="Open"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-1z7n62"
            >
              <span>
                Response type
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
  </div>
</div>
`;
